config/compilerFlags.cmake

```diff
@@ -5,6 +5,9 @@ if ( MINGW OR UNIX ) # MINGW, Linux, APPLE, CYGWIN
         set(COMPILER_IS_CLANG ON)
     endif()
 
+    set (CMAKE_CXX_FLAGS_RELEASE    "-O3 -DNDEBUG")
+    set (CMAKE_CXX_FLAGS_DEBUG      "-g -O0 -DDEBUG")

```

Comment:
How about `-g3 -gstrict-dwarf`, that includes macros in the debugging symbols (that is occasionally very handy)? Also gcc has `-Og` which is optimize for debug, but that already does some optimization (which can interfere with debugging).

And then a stupid question: what does the `DEBUG` macro do? Do we use that?

Reply:
I am happy to add `-g3 -gstrict-dwarf` but `-Og` affects to the debugging experience. What I do in some personal projects is to add a CMake variable (something like DEBUG_OPTIMIZATIONS) to control the addition of `-Og`. 

Then about the DEBUG definition, yes it is used in our code and it is actually used in many of the C++ implementation files. I actually should also check if I need to add it for Visual Studio, where the usage of that definition is more important.

Reply:
Wait ... I was wondering whether what I just said was right or not. It seems that VS adds _DEBUG when using some compiler flags, but DEBUG is not used in the internal implementation of C++. 

Anyways we are using the DEBUG definition in many places of the code, so we need to add it to enable some debug printing information.

Reply:
Yeah, just after I commented here, I saw the `#ifdef DEBUG ... #endif` in jp2image.cpp. Sorry, should have grep'ed for `DEBUG` first.

Concerning the optimization: you are right, lets keep it at `-O0`. Do you happen to know at which point the variable `CXXFLAGS` gets inserted? Since, at least for gcc, the last `-O` is the one that is effective if multiple ones are specified. So if the `CXXFLAGS` are pasted at the end, it would be very convenient for developers, as they would be able to override the optimization setting. But we can discus this in another issue.